### ISSUE TITLE
Option to close Osmose issues

### ISSUE THREAD

**Issue by jneidel** [Helium314/StreetComplete#686]
I'm using the Osmose quest.
My options are "Report false positive" or "Remove this issue locally".
I would like the ability to close an Osmose issue.

**Use case**

I came across all of these Osmose issues that I fixed in SCEE. I would have liked to close the Osmose issue instead of marking it as false positive (messes with it's analytics.)

- Unmapped max speed limit 30
- Unmapped weight restriction
- Unmapped animal_crossing
- and more

**Proposed Solution**

An extra answer option to close the issue.

**Comment by mnalis** [comment]
Yes, you should not mark it as false positive as that is not correct claim; but I wonder have you considered   _"Remove this issue locally"_, and if so, why doesn't it work for you?

After all, Osmose issues (if really resolved) should be _**automatically removed**_ on the next osmose run (which should be happening [daily](https://wiki.openstreetmap.org/wiki/Osmose#How_does_it_work?) according to the docs).

So yes, there is benefit that `Osmose close` would be hiding that Osmose issue from another (SCEE) users in the same area solving Osmose quests in less than 24 hours , but _only_ if they have not already downloaded the data before you've fixed the issue. Satisfying both  is rather small window, so I am not sure if it would improve situation much unless there is huge amount of very active Osmose mappers in your area (and adding extra answer is code to write and maintain, which might probably better be spent elsewhere). 
 
But for you personally, I think _"Remove this issue locally"_ should work equally well as would _"Close Osmose issue"_ API call. Given the above, do you agree, or have I missed something?

**Comment by jneidel** [comment]
I will test if "Remove this issue locally" actually closes/removes the issue. If that is the case, then the close option wouldn't be _necessary_.

When I read "Remove this issue locally", though, I think "this will hide the issue from _me_ in the future". Locally implies _to me_ that it is not synced to the server.
A "close" option would be would more clear. There would be no misunderstanding with that option in my mind.

**Comment by mnalis** [comment]
> Locally implies to me that it is not synced to the server.

Yeah, there is no API call to the **Osmose server** to change the Osmose issue status. Thus, it is only local removal of Osmose issue (i.e. it work equally well when you're offline).

But of course, any change to the **OSM tags** you make in SCEE (e.g. by solving a quest, or deleting a node, or manual tag edit, overlay changes etc) will result in OSM data being changed and uploaded to OSM API (that is the whole point of the SCEE, after all! If it never uploaded any data, it would be quite useless toy)

And Osmose will check OSM data from OSM servers daily, and detect which issues have been fixed since yesterday, as well as which new issues have occurred, and represent it in its database/map of issues. 

If you think about it, Osmose would be quite useless as automated checking tool if it didn't **automatically** detect both those fixes as well as regressions by itself (and instead if it needed humans to tell it manually for each new bug or each solved bug!)

---

Perhaps a better text / translation would help, but I can't seem to find short enough version (i.e. few words at most; and definitely much shorter than a few paragraphs above) that would be able to better clarify what it does / how it works. :man_shrugging:

**Comment by jneidel** [comment]
That does make sense. I'm just used to Vespuccis "close task" option on Osmose issues.

"Remove this issue locally" did not close/remove the issue on the server. It just hid it from me.

**Comment by mnalis** [comment]
> "Remove this issue locally" did not close/remove the issue on the server. It just hid it from me.

Not even after 24h hours have passed and Osmose re-verified its database? 

Can you link to example Osmose issue which had failed to automatically close after the underlying data which were causing the issue have been properly fixed in OSM database?

**Comment by goldfndr** [comment]
Clarification: within https://github.com/Helium314/SCEE/blob/modified/app/src/main/java/de/westnordost/streetcomplete/quests/osmose/OsmoseDao.kt at line 168, SCEE is coded to be able to upload "done" for a resolved issue, similar to Vespucci's PR 1651 in osmoseServer at changeState.

**Comment by esilja** [comment]
https://github.com/Helium314/SCEE/blob/8bc1d21e1581552634f370de4d55e9f8b0a8bffe/app/src/main/java/de/westnordost/streetcomplete/quests/osmose/OsmoseDao.kt#L168

https://github.com/MarcusWolschon/osmeditor4android/pull/1651

**Comment by jneidel** [comment]
As I understand it: on the edit of an item, a "done" update is sent to Osmose.

Wouldn't it make sense to hide the note (aka remove locally) for the user at the same time, so that they are not confused?

**Comment by mnalis** [comment]
> As I understand it: on the edit of an item, a "done" update is sent to Osmose.

Is that I guess or have you actually tested it? (I haven't)

> Wouldn't it make sense to hide the note (aka remove locally) for the user at the same time, so that they are not confused?

Perhaps (although I'm not sure even that editing an item should be sending "done", as there is no guarantee that editing has actually fixed the issue).  But I agree it would make sense to keep Osmose server and local state in sync (when possible).

However @jneidel that seems different issue then the one originally reported, if I am understanding it correctly?

For original issue, we'd still need an answer for https://github.com/Helium314/SCEE/issues/686#issuecomment-2464431092 if you could answer it?

**Comment by jneidel** [comment]
> Can you link to example Osmose issue which had failed to automatically close after the underlying data which were causing the issue have been properly fixed in OSM database?

I don't have an example for this.

**Comment by Helium314** [comment]
How it should behave (and actually behaves, at least last time I used it):
* "Remove this issue locally" removes the issue form the database it hides nothing, as in using the _hide_ functionality.
* Uploading any edit done in the Osmose quest marks the issue as _done_, which I think is what you mean by _close_. As long as you don't undo the edit, the issue will not be shown to you as Osmose quest. Possibly you can get the quest again when you download in the same area before uploading your changes, though.
Any Osmose issue set to done, but not uploaded, does not show up as quest.

**Comment by mnalis** [comment]
Seems to work as advertised.
- verified in https://osmose.openstreetmap.fr that some issue shows
- In SCEE 60.1 I've just e.g. removed `layer=1` in https://www.openstreetmap.org/way/29642348/history/10  
- a minute later I refreshed https://osmose.openstreetmap.fr and the issue no longer shows.

Thus, unless someone can produce example to the contrary, I'll close this as already implemented / works just fine (it is just not obvious that choosing to making edit from osmose quests marks it as _done_. If people are being confused by that, volunteers are invited to document it in the [SCEE FAQ](https://wiki.openstreetmap.org/wiki/SCEE/FAQ))
